products/views.py

```diff
@@ -0,0 +1,161 @@
+"""
+Contains Views for products app
+"""
+# pylint: disable=no-self-use, no-member
+from django.shortcuts import get_object_or_404
+from rest_framework import status, viewsets
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from .models import Product
+from .permissions import IsContentManager
+from .serializers import ProductSerializer
+
+
+class ProductViewSet(viewsets.ViewSet):
+    """
+    Product viewset
+    """
+
+    # pylint: disable =invalid-name
+    serializer_class = ProductSerializer
+    queryset = Product.objects.all()
+    authentication_classes = [TokenAuthentication]
+
+    def get_permissions(self):
+        """
+        Instantiates and returns the list of permissions that this view requires.
+        """
+        permission_classes = []
+        actions = ["create", "update", "delete"]
+        if self.action in actions:
+            permission_classes = [IsAuthenticated, IsContentManager]
+
+        return [permission() for permission in permission_classes]
+
+    def create(self, request):
+        """
+        Creates new Products
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (dict): Value containing product data
+        """
+        data = request.data.dict()
+        data["created_by"] = request.user.id
+        print(data)
+        serializer = ProductSerializer(data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(
+                {
+                    "message": "Product created successfully",
+                    "product_data": serializer.data,
+                    "status_code": status.HTTP_201_CREATED,
+                }
+            )
+        return Response(
+            {
+                "message": "Bad request",
+                "status_code": status.HTTP_400_BAD_REQUEST,
+                "errors": serializer.errors,
+            }
+        )
+
+    def list(self, request):
+        """
+        Returns all products
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (dict): Value containing all products
+        """
+        if request.method == "GET":
+            queryset = Product.objects.all()
+            if queryset:
+                return Response(
+                    {
+                        "products": ProductSerializer(queryset, many=True).data,
+                        "status_code": status.HTTP_200_OK,
+                    }
+                )
+            return Response(
+                {
+                    "message": "No products found",
+                    "status_code": status.HTTP_400_BAD_REQUEST,
+                }
+            )
+        return Response(
+            {"message": "Bad request", "status_code": status.HTTP_400_BAD_REQUEST}
```

Comment:
What's the reason for overriding this method?

Reply:
This method returns all the products in database. It helps in changing a product or changing it(by providing primary key). it also helps seeing the changes or deletions. Also it might be useful when we will create a front end. Maybe we can add some customization/filters to it to make it more practical

Reply:
I mean in understand, My point was regarding the need to override it instead of using the default list behavior? Usually, it is a good idea to not do what we don't need.

Reply:
Oh ok I get it now. Thanks for this comment, it really made me see the advantage of using ModelViewSets. I have updated the `views.py` file